### ISSUE TITLE
Elasticsearch bulk index failed

### ISSUE THREAD

**Issue by lovelgw** [elastic/elasticsearch#22347]
**Elasticsearch version**: 5.1.1

**Plugins installed**:  analysis-seunjeon, elasticsearch-analysis-mecab-ko, x-pack

**JVM version**: openjdk version "1.8.0_111"

**OS version**: CentOS release 6.8 (Final)

**Description of the problem including expected versus actual behavior**:
bulk index fails 

**Steps to reproduce**:
 1. curl -s -XPOST http://127.0.0.1:9200/data/_bulk --data-binary '@test.json'

**Provide logs (if relevant)**:
[2016-12-26T11:39:54,182][DEBUG][o.e.a.b.TransportShardBulkAction] [NODE_175] [data][4] failed to execute bulk item (index) index {[data][paper][1723759], source[{"docSeq":1723759,"memberSeq":801362,"memberId":"kstudy","docCategorySeq":451,"title":"\uc81c 1 \ubd80 \ubab8\uc9d3 \uc5b8\uc5b4\uc640 \uae30\ud638\ud559 : \ud55c\uad6d\uc778\uc758 \uc5b8\uc5b4 \uc608\uc808\uacfc \uc2e0\uccb4 \uc5b8\uc5b4 (1 . Body Language and Semiotics : Korean Language - etiquettes and Body - language)","titleNoSpace":"\uc81c1\ubd80\ubab8\uc9d3\uc5b8\uc5b4\uc640\uae30\ud638\ud559:\ud55c\uad6d\uc778\uc758\uc5b8\uc5b4\uc608\uc808\uacfc\uc2e0\uccb4\uc5b8\uc5b4(1.BodyLanguageandSemiotics:KoreanLanguage-etiquettesandBody-language)","tag":["\ud55c\uad6d\uc778","\uc5b8\uc5b4\uc608\uc808","\uc2e0\uccb4\uc5b8\uc5b4","\uc5b8\uc5b4\uc640\uae30","\ubd80\ubab8\uc9d3","\ud638\ud559\ud55c\uad6d","\uc808\uacfc\uc2e0\uccb4"],"ext":"pdf","price":6800,"page":38,"ratingPoint":0,"contents":"","body":"  \ud55c\uad6d\uae30\ud638\ud559\ud68c \uae30\ud638\ud559 \uc5f0\uad6c \uc131\uad11\uc218(Kwang Su Sung),\uae40\uc131\ub3c4(Sung Do Kim)","sCategory":2,"bCategory":439,"category":"PP105","uploadDate":1057824810,"regDate":1057824810,"changeDate":0,"originalCategory":"1057","reference":"","knouCourseSeq":0,"year":0,"knouExaminationSeq":0,"grade":0,"term":0,"courseCode":0,"knouDepartmentSeq":0,"viewCount":0,"scrapCount":0,"useMobile":"N","knouReportTypeSeq":0,"knouReportTypeName":"","knouDepartmentName":"","knouCourseName":"","lessonName":"","keyword":[],"searchKeyword":["\uc81c 1 \ubd80 \ubab8\uc9d3 \uc5b8\uc5b4\uc640 \uae30\ud638\ud559 : \ud55c\uad6d\uc778\uc758 \uc5b8\uc5b4 \uc608\uc808\uacfc \uc2e0\uccb4 \uc5b8\uc5b4"],"matchKeyword":["\uc81c 1 \ubd80 \ubab8\uc9d3 \uc5b8\uc5b4\uc640 \uae30\ud638\ud559 : \ud55c\uad6d\uc778\uc758 \uc5b8\uc5b4 \uc608\uc808\uacfc \uc2e0\uccb4 \uc5b8\uc5b4"],"saleEventStartDate":0,"saleEventEndDate":0,"discountPercent":0,"discountPrice":0}]}
java.lang.IllegalArgumentException: automaton has cycles
	at org.apache.lucene.util.automaton.FiniteStringsIterator.next(FiniteStringsIterator.java:121) ~[lucene-core-6.3.0.jar:6.3.0 a66a44513ee8191e25b477372094bfa846450316 - shalin - 2016-11-02 19:47:11]
	at org.apache.lucene.util.automaton.LimitedFiniteStringsIterator.next(LimitedFiniteStringsIterator.java:69) ~[lucene-core-6.3.0.jar:6.3.0 a66a44513ee8191e25b477372094bfa846450316 - shalin - 2016-11-02 19:47:11]
	at org.apache.lucene.search.suggest.document.CompletionTokenStream.incrementToken(CompletionTokenStream.java:103) ~[lucene-suggest-6.3.0.jar:6.3.0 a66a44513ee8191e25b477372094bfa846450316 - shalin - 2016-11-02 19:48:02]
	at org.apache.lucene.index.DefaultIndexingChain$PerField.invert(DefaultIndexingChain.java:712) ~[lucene-core-6.3.0.jar:6.3.0 a66a44513ee8191e25b477372094bfa846450316 - shalin - 2016-11-02 19:47:11]
	at org.apache.lucene.index.DefaultIndexingChain.processField(DefaultIndexingChain.java:417) ~[lucene-core-6.3.0.jar:6.3.0 a66a44513ee8191e25b477372094bfa846450316 - shalin - 2016-11-02 19:47:11]
	at org.apache.lucene.index.DefaultIndexingChain.processDocument(DefaultIndexingChain.java:373) ~[lucene-core-6.3.0.jar:6.3.0 a66a44513ee8191e25b477372094bfa846450316 - shalin - 2016-11-02 19:47:11]
	at org.apache.lucene.index.DocumentsWriterPerThread.updateDocument(DocumentsWriterPerThread.java:231) ~[lucene-core-6.3.0.jar:6.3.0 a66a44513ee8191e25b477372094bfa846450316 - shalin - 2016-11-02 19:47:11]
	at org.apache.lucene.index.DocumentsWriter.updateDocument(DocumentsWriter.java:478) ~[lucene-core-6.3.0.jar:6.3.0 a66a44513ee8191e25b477372094bfa846450316 - shalin - 2016-11-02 19:47:11]
	at org.apache.lucene.index.IndexWriter.updateDocument(IndexWriter.java:1562) ~[lucene-core-6.3.0.jar:6.3.0 a66a44513ee8191e25b477372094bfa846450316 - shalin - 2016-11-02 19:47:11]
	at org.apache.lucene.index.IndexWriter.addDocument(IndexWriter.java:1307) ~[lucene-core-6.3.0.jar:6.3.0 a66a44513ee8191e25b477372094bfa846450316 - shalin - 2016-11-02 19:47:11]
	at org.elasticsearch.index.engine.InternalEngine.index(InternalEngine.java:558) ~[elasticsearch-5.1.1.jar:5.1.1]
	at org.elasticsearch.index.engine.InternalEngine.innerIndex(InternalEngine.java:520) ~[elasticsearch-5.1.1.jar:5.1.1]
	at org.elasticsearch.index.engine.InternalEngine.index(InternalEngine.java:409) ~[elasticsearch-5.1.1.jar:5.1.1]
	at org.elasticsearch.index.shard.IndexShard.index(IndexShard.java:556) ~[elasticsearch-5.1.1.jar:5.1.1]
	at org.elasticsearch.index.shard.IndexShard.index(IndexShard.java:546) ~[elasticsearch-5.1.1.jar:5.1.1]
	at org.elasticsearch.action.index.TransportIndexAction.executeIndexRequestOnPrimary(TransportIndexAction.java:191) ~[elasticsearch-5.1.1.jar:5.1.1]
	at org.elasticsearch.action.bulk.TransportShardBulkAction.shardIndexOperation(TransportShardBulkAction.java:348) ~[elasticsearch-5.1.1.jar:5.1.1]
	at org.elasticsearch.action.bulk.TransportShardBulkAction.index(TransportShardBulkAction.java:155) ~[elasticsearch-5.1.1.jar:5.1.1]
	at org.elasticsearch.action.bulk.TransportShardBulkAction.handleItem(TransportShardBulkAction.java:134) ~[elasticsearch-5.1.1.jar:5.1.1]
	at org.elasticsearch.action.bulk.TransportShardBulkAction.onPrimaryShard(TransportShardBulkAction.java:120) ~[elasticsearch-5.1.1.jar:5.1.1]
	at org.elasticsearch.action.bulk.TransportShardBulkAction.onPrimaryShard(TransportShardBulkAction.java:73) ~[elasticsearch-5.1.1.jar:5.1.1]
	at org.elasticsearch.action.support.replication.TransportWriteAction.shardOperationOnPrimary(TransportWriteAction.java:75) ~[elasticsearch-5.1.1.jar:5.1.1]
	at org.elasticsearch.action.support.replication.TransportWriteAction.shardOperationOnPrimary(TransportWriteAction.java:48) ~[elasticsearch-5.1.1.jar:5.1.1]
	at org.elasticsearch.action.support.replication.TransportReplicationAction$PrimaryShardReference.perform(TransportReplicationAction.java:905) ~[elasticsearch-5.1.1.jar:5.1.1]
	at org.elasticsearch.action.support.replication.TransportReplicationAction$PrimaryShardReference.perform(TransportReplicationAction.java:875) ~[elasticsearch-5.1.1.jar:5.1.1]
	at org.elasticsearch.action.support.replication.ReplicationOperation.execute(ReplicationOperation.java:113) ~[elasticsearch-5.1.1.jar:5.1.1]
	at org.elasticsearch.action.support.replication.TransportReplicationAction$AsyncPrimaryAction.onResponse(TransportReplicationAction.java:323) ~[elasticsearch-5.1.1.jar:5.1.1]
	at org.elasticsearch.action.support.replication.TransportReplicationAction$AsyncPrimaryAction.onResponse(TransportReplicationAction.java:258) ~[elasticsearch-5.1.1.jar:5.1.1]
	at org.elasticsearch.action.support.replication.TransportReplicationAction$1.onResponse(TransportReplicationAction.java:855) ~[elasticsearch-5.1.1.jar:5.1.1]
	at org.elasticsearch.action.support.replication.TransportReplicationAction$1.onResponse(TransportReplicationAction.java:852) ~[elasticsearch-5.1.1.jar:5.1.1]
	at org.elasticsearch.index.shard.IndexShardOperationsLock.acquire(IndexShardOperationsLock.java:142) ~[elasticsearch-5.1.1.jar:5.1.1]
	at org.elasticsearch.index.shard.IndexShard.acquirePrimaryOperationLock(IndexShard.java:1655) ~[elasticsearch-5.1.1.jar:5.1.1]
	at org.elasticsearch.action.support.replication.TransportReplicationAction.acquirePrimaryShardReference(TransportReplicationAction.java:864) ~[elasticsearch-5.1.1.jar:5.1.1]
	at org.elasticsearch.action.support.replication.TransportReplicationAction.access$400(TransportReplicationAction.java:90) ~[elasticsearch-5.1.1.jar:5.1.1]
	at org.elasticsearch.action.support.replication.TransportReplicationAction$AsyncPrimaryAction.doRun(TransportReplicationAction.java:275) ~[elasticsearch-5.1.1.jar:5.1.1]
	at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37) ~[elasticsearch-5.1.1.jar:5.1.1]
	at org.elasticsearch.action.support.replication.TransportReplicationAction$PrimaryOperationTransportHandler.messageReceived(TransportReplicationAction.java:254) ~[elasticsearch-5.1.1.jar:5.1.1]
	at org.elasticsearch.action.support.replication.TransportReplicationAction$PrimaryOperationTransportHandler.messageReceived(TransportReplicationAction.java:246) ~[elasticsearch-5.1.1.jar:5.1.1]
	at org.elasticsearch.xpack.security.transport.SecurityServerTransportInterceptor$ProfileSecuredRequestHandler.lambda$messageReceived$1(SecurityServerTransportInterceptor.java:208) ~[x-pack-5.1.1.jar:5.1.1]
	at org.elasticsearch.xpack.security.transport.SecurityServerTransportInterceptor$ProfileSecuredRequestHandler.messageReceived(SecurityServerTransportInterceptor.java:255) [x-pack-5.1.1.jar:5.1.1]
	at org.elasticsearch.transport.RequestHandlerRegistry.processMessageReceived(RequestHandlerRegistry.java:69) [elasticsearch-5.1.1.jar:5.1.1]
	at org.elasticsearch.transport.TransportService$6.doRun(TransportService.java:577) [elasticsearch-5.1.1.jar:5.1.1]
	at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:527) [elasticsearch-5.1.1.jar:5.1.1]
	at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37) [elasticsearch-5.1.1.jar:5.1.1]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) [?:1.8.0_111]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) [?:1.8.0_111]
	at java.lang.Thread.run(Thread.java:745) [?:1.8.0_111]

**Comment by ywelsch** [comment]
> java.lang.IllegalArgumentException: automaton has cycles
at org.apache.lucene.util.automaton.FiniteStringsIterator.next(FiniteStringsIterator.java:121) ~[lucene-core-6.3.0.jar:6.3.0 a66a44513ee8191e25b477372094bfa846450316 - shalin - 2016-11-02 19:47:11]

Can you provide the mapping for the corresponding index? (i.e. output of `curl localhost:9200/data/_mapping/paper?pretty`)

/cc: @mikemccand

**Comment by lovelgw** [comment]
{
  "data": {
    "mappings": {
      "_default_": {
        "properties": {
          "bCategory": {
            "type": "integer",
            "store": true
          },
          "body": {
            "type": "text",
            "analyzer": "koreanIndex"
          },
          "category": {
            "type": "keyword",
            "store": true
          },
          "changeDate": {
            "type": "date",
            "format": "epoch_second"
          },
          "contents": {
            "type": "text",
            "analyzer": "koreanIndex"
          },
          "courseCode": {
            "type": "integer",
            "store": true
          },
          "discountPercent": {
            "type": "integer",
            "store": true,
            "null_value": 0
          },
          "discountPrice": {
            "type": "integer",
            "store": true,
            "null_value": 0
          },
          "docCategorySeq": {
            "type": "integer",
            "store": true
          },
          "docSeq": {
            "type": "integer",
            "store": true
          },
          "ext": {
            "type": "keyword",
            "store": true
          },
          "grade": {
            "type": "integer",
            "store": true
          },
          "knouCourseName": {
            "type": "keyword",
            "store": true
          },
          "knouCourseSeq": {
            "type": "integer",
            "store": true
          },
          "knouDepartmentName": {
            "type": "keyword",
            "store": true
          },
          "knouDepartmentSeq": {
            "type": "integer",
            "store": true
          },
          "knouExaminationSeq": {
            "type": "integer",
            "store": true
          },
          "knouReportTypeName": {
            "type": "keyword",
            "store": true
          },
          "knouReportTypeSeq": {
            "type": "integer",
            "store": true
          },
          "lessonName": {
            "type": "keyword",
            "store": true
          },
          "matchKeyword": {
            "type": "text",
            "analyzer": "koreanIndex"
          },
          "memberId": {
            "type": "keyword",
            "store": true
          },
          "memberSeq": {
            "type": "integer",
            "store": true
          },
          "originalCategory": {
            "type": "keyword",
            "store": true
          },
          "page": {
            "type": "integer",
            "store": true
          },
          "price": {
            "type": "integer",
            "store": true
          },
          "ratingPoint": {
            "type": "integer",
            "store": true
          },
          "reference": {
            "type": "text",
            "analyzer": "koreanIndex"
          },
          "regDate": {
            "type": "date",
            "format": "epoch_second"
          },
          "sCategory": {
            "type": "integer",
            "store": true
          },
          "saleEventEndDate": {
            "type": "date",
            "format": "epoch_second"
          },
          "saleEventStartDate": {
            "type": "date",
            "format": "epoch_second"
          },
          "scrapCount": {
            "type": "integer",
            "store": true
          },
          "searchKeyword": {
            "type": "completion",
            "analyzer": "koreanIndex",
            "preserve_separators": true,
            "preserve_position_increments": true,
            "max_input_length": 50
          },
          "tag": {
            "type": "text",
            "store": true,
            "analyzer": "koreanIndex"
          },
          "term": {
            "type": "integer",
            "store": true
          },
          "title": {
            "type": "text",
            "analyzer": "koreanIndex"
          },
          "titleNoSpace": {
            "type": "text"
          },
          "uploadDate": {
            "type": "date",
            "format": "epoch_second"
          },
          "useMobile": {
            "type": "keyword",
            "store": true
          },
          "viewCount": {
            "type": "integer",
            "store": true
          },
          "year": {
            "type": "integer",
            "store": true
          }
        }
      }
    }
  }
}

**Comment by mikemccand** [comment]
Wow this is a curious exception ... can you post your `koreanIndex` analysis chain?  With that, and with the string you are trying to index into the `searchKeyword` field (which I see above), hopefully I can make a failing Lucene test case.

What's happening is the completion analyzer needs to turn the incoming tokens into an automaton and then enumerate all paths through it, but somehow a cyclic automaton is being created, which should not be possible from any token stream.

**Comment by lovelgw** [comment]
koreanIndex analyzer result 

**title field :  "제 1 부 몸짓 언어와 기호학 : 한국인의 언어 예절과 신체 언어 (1 . Body Language and Semiotics : Korean Language - etiquettes and Body - language)"
{
  "tokens" : [
    {
      "token" : "제",
      "start_offset" : 0,
      "end_offset" : 1,
      "type" : "XPN",
      "position" : 0
    },
    {
      "token" : "1",
      "start_offset" : 2,
      "end_offset" : 3,
      "type" : "SN",
      "position" : 1
    },
    {
      "token" : "부",
      "start_offset" : 4,
      "end_offset" : 5,
      "type" : "NNBC",
      "position" : 2
    },
    {
      "token" : "몸짓",
      "start_offset" : 6,
      "end_offset" : 8,
      "type" : "COMPOUND",
      "position" : 3
    },
    {
      "token" : "언어와",
      "start_offset" : 9,
      "end_offset" : 12,
      "type" : "EOJEOL",
      "position" : 4
    },
    {
      "token" : "언어",
      "start_offset" : 9,
      "end_offset" : 11,
      "type" : "NNG",
      "position" : 4
    },
    {
      "token" : "기호",
      "start_offset" : 13,
      "end_offset" : 15,
      "type" : "NNG",
      "position" : 5
    },
    {
      "token" : "기호학",
      "start_offset" : 13,
      "end_offset" : 16,
      "type" : "COMPOUND",
      "position" : 5
    },
    {
      "token" : "학",
      "start_offset" : 15,
      "end_offset" : 16,
      "type" : "NNG",
      "position" : 6
    },
    {
      "token" : "한국",
      "start_offset" : 19,
      "end_offset" : 21,
      "type" : "NNG",
      "position" : 7
    },
    {
      "token" : "한국인의",
      "start_offset" : 19,
      "end_offset" : 23,
      "type" : "EOJEOL",
      "position" : 7
    },
    {
      "token" : "한국인",
      "start_offset" : 19,
      "end_offset" : 22,
      "type" : "COMPOUND",
      "position" : 7
    },
    {
      "token" : "인",
      "start_offset" : 21,
      "end_offset" : 22,
      "type" : "NNG",
      "position" : 8
    },
    {
      "token" : "언어",
      "start_offset" : 24,
      "end_offset" : 26,
      "type" : "NNG",
      "position" : 9
    },
    {
      "token" : "예절과",
      "start_offset" : 27,
      "end_offset" : 30,
      "type" : "EOJEOL",
      "position" : 10
    },
    {
      "token" : "예절",
      "start_offset" : 27,
      "end_offset" : 29,
      "type" : "NNG",
      "position" : 10
    },
    {
      "token" : "신체",
      "start_offset" : 31,
      "end_offset" : 33,
      "type" : "NNG",
      "position" : 11
    },
    {
      "token" : "언어",
      "start_offset" : 34,
      "end_offset" : 36,
      "type" : "NNG",
      "position" : 12
    },
    {
      "token" : "1",
      "start_offset" : 38,
      "end_offset" : 39,
      "type" : "SN",
      "position" : 13
    },
    {
      "token" : "body",
      "start_offset" : 42,
      "end_offset" : 46,
      "type" : "SL",
      "position" : 14
    },
    {
      "token" : "language",
      "start_offset" : 47,
      "end_offset" : 55,
      "type" : "SL",
      "position" : 15
    },
    {
      "token" : "and",
      "start_offset" : 56,
      "end_offset" : 59,
      "type" : "SL",
      "position" : 16
    },
    {
      "token" : "semiotics",
      "start_offset" : 60,
      "end_offset" : 69,
      "type" : "SL",
      "position" : 17
    },
    {
      "token" : "korean",
      "start_offset" : 72,
      "end_offset" : 78,
      "type" : "SL",
      "position" : 18
    },
    {
      "token" : "language",
      "start_offset" : 79,
      "end_offset" : 87,
      "type" : "SL",
      "position" : 19
    },
    {
      "token" : "etiquettes",
      "start_offset" : 90,
      "end_offset" : 100,
      "type" : "SL",
      "position" : 20
    },
    {
      "token" : "and",
      "start_offset" : 101,
      "end_offset" : 104,
      "type" : "SL",
      "position" : 21
    },
    {
      "token" : "body",
      "start_offset" : 105,
      "end_offset" : 109,
      "type" : "SL",
      "position" : 22
    },
    {
      "token" : "language",
      "start_offset" : 112,
      "end_offset" : 120,
      "type" : "SL",
      "position" : 23
    }
  ]
}


titlenospace : "제1부몸짓언어와기호학:한국인의언어예절과신체언어(1.BodyLanguageandSemiotics:KoreanLanguage-etiquettesandBody-language)"
{
  "tokens" : [
    {
      "token" : "제",
      "start_offset" : 0,
      "end_offset" : 1,
      "type" : "XPN",
      "position" : 0
    },
    {
      "token" : "1",
      "start_offset" : 1,
      "end_offset" : 2,
      "type" : "SN",
      "position" : 1
    },
    {
      "token" : "부",
      "start_offset" : 2,
      "end_offset" : 3,
      "type" : "NNBC",
      "position" : 2
    },
    {
      "token" : "몸짓",
      "start_offset" : 3,
      "end_offset" : 5,
      "type" : "COMPOUND",
      "position" : 3
    },
    {
      "token" : "언어와",
      "start_offset" : 5,
      "end_offset" : 8,
      "type" : "EOJEOL",
      "position" : 4
    },
    {
      "token" : "언어",
      "start_offset" : 5,
      "end_offset" : 7,
      "type" : "NNG",
      "position" : 4
    },
    {
      "token" : "기호",
      "start_offset" : 8,
      "end_offset" : 10,
      "type" : "NNG",
      "position" : 5
    },
    {
      "token" : "기호학",
      "start_offset" : 8,
      "end_offset" : 11,
      "type" : "COMPOUND",
      "position" : 5
    },
    {
      "token" : "학",
      "start_offset" : 10,
      "end_offset" : 11,
      "type" : "NNG",
      "position" : 6
    },
    {
      "token" : "한국",
      "start_offset" : 12,
      "end_offset" : 14,
      "type" : "NNG",
      "position" : 7
    },
    {
      "token" : "한국인의",
      "start_offset" : 12,
      "end_offset" : 16,
      "type" : "EOJEOL",
      "position" : 7
    },
    {
      "token" : "한국인",
      "start_offset" : 12,
      "end_offset" : 15,
      "type" : "COMPOUND",
      "position" : 7
    },
    {
      "token" : "인",
      "start_offset" : 14,
      "end_offset" : 15,
      "type" : "NNG",
      "position" : 8
    },
    {
      "token" : "언어",
      "start_offset" : 16,
      "end_offset" : 18,
      "type" : "NNG",
      "position" : 9
    },
    {
      "token" : "예절과",
      "start_offset" : 18,
      "end_offset" : 21,
      "type" : "EOJEOL",
      "position" : 10
    },
    {
      "token" : "예절",
      "start_offset" : 18,
      "end_offset" : 20,
      "type" : "NNG",
      "position" : 10
    },
    {
      "token" : "신체",
      "start_offset" : 21,
      "end_offset" : 23,
      "type" : "NNG",
      "position" : 11
    },
    {
      "token" : "언어",
      "start_offset" : 23,
      "end_offset" : 25,
      "type" : "NNG",
      "position" : 12
    },
    {
      "token" : "1",
      "start_offset" : 26,
      "end_offset" : 27,
      "type" : "SN",
      "position" : 13
    },
    {
      "token" : "bodylanguageandsemiotics",
      "start_offset" : 28,
      "end_offset" : 52,
      "type" : "SL",
      "position" : 14
    },
    {
      "token" : "koreanlanguage",
      "start_offset" : 53,
      "end_offset" : 67,
      "type" : "SL",
      "position" : 15
    },
    {
      "token" : "etiquettesandbody",
      "start_offset" : 68,
      "end_offset" : 85,
      "type" : "SL",
      "position" : 16
    },
    {
      "token" : "language",
      "start_offset" : 86,
      "end_offset" : 94,
      "type" : "SL",
      "position" : 17
    }
  ]
}


body :  " 한국기호학회 기호학 연구 성광수(Kwang Su Sung),김성도(Sung Do Kim)"
{
  "tokens" : [
    {
      "token" : "한국",
      "start_offset" : 1,
      "end_offset" : 3,
      "type" : "NNP",
      "position" : 0
    },
    {
      "token" : "기호",
      "start_offset" : 3,
      "end_offset" : 5,
      "type" : "NNG",
      "position" : 1
    },
    {
      "token" : "학회",
      "start_offset" : 5,
      "end_offset" : 7,
      "type" : "NNG",
      "position" : 2
    },
    {
      "token" : "기호",
      "start_offset" : 8,
      "end_offset" : 10,
      "type" : "NNG",
      "position" : 3
    },
    {
      "token" : "기호학",
      "start_offset" : 8,
      "end_offset" : 11,
      "type" : "COMPOUND",
      "position" : 3
    },
    {
      "token" : "학",
      "start_offset" : 10,
      "end_offset" : 11,
      "type" : "NNG",
      "position" : 4
    },
    {
      "token" : "연구",
      "start_offset" : 12,
      "end_offset" : 14,
      "type" : "NNG",
      "position" : 5
    },
    {
      "token" : "성광수",
      "start_offset" : 15,
      "end_offset" : 18,
      "type" : "NNP",
      "position" : 6
    },
    {
      "token" : "kwang",
      "start_offset" : 19,
      "end_offset" : 24,
      "type" : "SL",
      "position" : 7
    },
    {
      "token" : "su",
      "start_offset" : 25,
      "end_offset" : 27,
      "type" : "SL",
      "position" : 8
    },
    {
      "token" : "sung",
      "start_offset" : 28,
      "end_offset" : 32,
      "type" : "SL",
      "position" : 9
    },
    {
      "token" : "김성도",
      "start_offset" : 34,
      "end_offset" : 37,
      "type" : "NNP",
      "position" : 10
    },
    {
      "token" : "sung",
      "start_offset" : 38,
      "end_offset" : 42,
      "type" : "SL",
      "position" : 11
    },
    {
      "token" : "do",
      "start_offset" : 43,
      "end_offset" : 45,
      "type" : "SL",
      "position" : 12
    },
    {
      "token" : "kim",
      "start_offset" : 46,
      "end_offset" : 49,
      "type" : "SL",
      "position" : 13
    }
  ]
}


searchKeyword[0] , matchKeyword[0] :  "제 1 부 몸짓 언어와 기호학 : 한국인의 언어 예절과 신체 언어"
{
  "tokens" : [
    {
      "token" : "제",
      "start_offset" : 0,
      "end_offset" : 1,
      "type" : "XPN",
      "position" : 0
    },
    {
      "token" : "1",
      "start_offset" : 2,
      "end_offset" : 3,
      "type" : "SN",
      "position" : 1
    },
    {
      "token" : "부",
      "start_offset" : 4,
      "end_offset" : 5,
      "type" : "NNBC",
      "position" : 2
    },
    {
      "token" : "몸짓",
      "start_offset" : 6,
      "end_offset" : 8,
      "type" : "COMPOUND",
      "position" : 3
    },
    {
      "token" : "언어와",
      "start_offset" : 9,
      "end_offset" : 12,
      "type" : "EOJEOL",
      "position" : 4
    },
    {
      "token" : "언어",
      "start_offset" : 9,
      "end_offset" : 11,
      "type" : "NNG",
      "position" : 4
    },
    {
      "token" : "기호",
      "start_offset" : 13,
      "end_offset" : 15,
      "type" : "NNG",
      "position" : 5
    },
    {
      "token" : "기호학",
      "start_offset" : 13,
      "end_offset" : 16,
      "type" : "COMPOUND",
      "position" : 5
    },
    {
      "token" : "학",
      "start_offset" : 15,
      "end_offset" : 16,
      "type" : "NNG",
      "position" : 6
    },
    {
      "token" : "한국",
      "start_offset" : 19,
      "end_offset" : 21,
      "type" : "NNG",
      "position" : 7
    },
    {
      "token" : "한국인의",
      "start_offset" : 19,
      "end_offset" : 23,
      "type" : "EOJEOL",
      "position" : 7
    },
    {
      "token" : "한국인",
      "start_offset" : 19,
      "end_offset" : 22,
      "type" : "COMPOUND",
      "position" : 7
    },
    {
      "token" : "인",
      "start_offset" : 21,
      "end_offset" : 22,
      "type" : "NNG",
      "position" : 8
    },
    {
      "token" : "언어",
      "start_offset" : 24,
      "end_offset" : 26,
      "type" : "NNG",
      "position" : 9
    },
    {
      "token" : "예절과",
      "start_offset" : 27,
      "end_offset" : 30,
      "type" : "EOJEOL",
      "position" : 10
    },
    {
      "token" : "예절",
      "start_offset" : 27,
      "end_offset" : 29,
      "type" : "NNG",
      "position" : 10
    },
    {
      "token" : "신체",
      "start_offset" : 31,
      "end_offset" : 33,
      "type" : "NNG",
      "position" : 11
    },
    {
      "token" : "언어",
      "start_offset" : 34,
      "end_offset" : 36,
      "type" : "NNG",
      "position" : 12
    }
  ]
}


tag[0] : "한국인"
{
  "tokens" : [
    {
      "token" : "한국",
      "start_offset" : 0,
      "end_offset" : 2,
      "type" : "NNG",
      "position" : 0
    },
    {
      "token" : "한국인",
      "start_offset" : 0,
      "end_offset" : 3,
      "type" : "COMPOUND",
      "position" : 0
    },
    {
      "token" : "인",
      "start_offset" : 2,
      "end_offset" : 3,
      "type" : "NNG",
      "position" : 1
    }
  ]
}

tag[1] : "언어예절"
{
  "tokens" : [
    {
      "token" : "언어",
      "start_offset" : 0,
      "end_offset" : 2,
      "type" : "NNG",
      "position" : 0
    },
    {
      "token" : "예절",
      "start_offset" : 2,
      "end_offset" : 4,
      "type" : "NNG",
      "position" : 1
    }
  ]
}

tag[2] : "신체언어"
{
  "tokens" : [
    {
      "token" : "신체",
      "start_offset" : 0,
      "end_offset" : 2,
      "type" : "NNG",
      "position" : 0
    },
    {
      "token" : "언어",
      "start_offset" : 2,
      "end_offset" : 4,
      "type" : "NNG",
      "position" : 1
    }
  ]
}
tag[3] : "언어와기"
{
  "tokens" : [
    {
      "token" : "언어와",
      "start_offset" : 0,
      "end_offset" : 3,
      "type" : "EOJEOL",
      "position" : 0
    },
    {
      "token" : "언어",
      "start_offset" : 0,
      "end_offset" : 2,
      "type" : "NNG",
      "position" : 0
    },
    {
      "token" : "기",
      "start_offset" : 3,
      "end_offset" : 4,
      "type" : "E",
      "position" : 1
    }
  ]
}
tag[4] : "부몸짓"
{
  "tokens" : [
    {
      "token" : "부몸짓",
      "start_offset" : 0,
      "end_offset" : 3,
      "type" : "EOJEOL",
      "position" : 0
    },
    {
      "token" : "부",
      "start_offset" : 0,
      "end_offset" : 1,
      "type" : "XPN",
      "position" : 0
    },
    {
      "token" : "몸짓",
      "start_offset" : 1,
      "end_offset" : 3,
      "type" : "COMPOUND",
      "position" : 0
    }
  ]
}
tag[5] : "호학한국"
{
  "tokens" : [
    {
      "token" : "호학",
      "start_offset" : 0,
      "end_offset" : 2,
      "type" : "NNG",
      "position" : 0
    },
    {
      "token" : "한국",
      "start_offset" : 2,
      "end_offset" : 4,
      "type" : "NNP",
      "position" : 1
    }
  ]
}

tag[6] : "절과신체"
{
  "tokens" : [
    {
      "token" : "절과",
      "start_offset" : 0,
      "end_offset" : 2,
      "type" : "EOJEOL",
      "position" : 0
    },
    {
      "token" : "절",
      "start_offset" : 0,
      "end_offset" : 1,
      "type" : "NNG",
      "position" : 0
    },
    {
      "token" : "신체",
      "start_offset" : 2,
      "end_offset" : 4,
      "type" : "NNG",
      "position" : 1
    }
  ]
}**

**Comment by mikemccand** [comment]
Thank you @lovelgw but I need to know how you configured your `koreanIndex` analyzer so I can also create those tokens from the input string in a Lucene test case.

**Comment by lovelgw** [comment]
settings and mappings used to create the index
[index_create.txt](https://github.com/elastic/elasticsearch/files/682210/index_create.txt)

**Comment by mikemccand** [comment]
Hmm, `seunjeon_tokenizer` ... I'll need to look at its sources to see how it's creating a cyclic graph.

Thanks @lovelgw

**Comment by mikemccand** [comment]
This seems to be the source code for the the `seunjeon_tokenizer` plugin: https://bitbucket.org/eunjeon/seunjeon/src/f7775cf0afd6ba7b7cc336214700dcdba2b1918a/elasticsearch/?at=master

**Comment by mikemccand** [comment]
OK the `SeunjeonTokenizer` sources have this line:

```
            posLenAtt.setPositionLength(pos.positionLength());
```

I strongly suspect that for your string, the position length is 0, which is an illegal value, but Lucene's `PackedTokenAttributeImpl.java` failed to catch that (I've now fixed that, for Lucene's next (6.4) release).

I think at this point you should open an issue/PR against the `seunjeon` project; there is nothing ES can do when the tokenizer produces an invalid position length.
